helix-core/src/main/java/org/apache/helix/model/PartitionWithReplicaCount.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.helix.model;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Objects;
+
+
+/**
+ * A POJO class containing {@link Partition} with missing replicas.
+ */
+public class PartitionWithReplicaCount {
```

Comment:
NIT: Why we need this object? They are nested partitions in the List object. The only need is get replica count. We can store list Partition or just partition name as a String. When we compute, we can get MinActiveReplica easily, right.  Also equals is not necessary here, right?

Here I am a little bit worrying about the too many objects can cause GCs for each of pipeline as these objects will not be hold in memory and continuously update.

Reply:
yeah I can replace the Partition with string name only. 

I created this class to reduce some duplication and overhead especially on computing minActiveReplica, it's cleaner to bundle things together.
Regarding GC, it's non-static, it's not holding too many things (only partitions that failed minActiveReplica), and it's only held within the method invocation of `findToBeAssignedReplicasForMinActiveReplica`. I think we can give GC a chance.

Reply:
I got your point but still since we do not need to pass the object around. Thus create a class just for small step of intermediate computations would be a little bit over. Highly recommend not have this class.

Reply:
Ok, will change. Thanks.